desc/integrals/quad_utils.py

```diff
@@ -190,14 +184,67 @@ def leggauss_lob(deg, interior_only=False):
     return x, w
 
 
+def chebgauss_uniform(deg):
+    """Gauss-Chebyshev quadrature with uniformly spaced nodes.
+
+    Returns quadrature points xₖ and weights wₖ for the approximate evaluation
+    of the integral ∫₋₁¹ f(x) dx ≈ ∑ₖ wₖ f(xₖ).
+
+    Parameters
+    ----------
+    deg : int
+        Number of quadrature points.
+
+    Returns
+    -------
+    x, w : (jnp.ndarray, jnp.ndarray)
+        Shape (deg, ).
+        Quadrature points and weights.
+
+    """
+    # Define x = 2/π arcsin y and g : y ↦ f(x(y)).
+    #   ∫₋₁¹ f(x) dx = 2/π ∫₋₁¹ (1−y²)⁻⁰ᐧ⁵ g(y) dy
+    # ∑ₖ wₖ f(x(yₖ)) = 2/π ∑ₖ ωₖ g(yₖ)
+    # Given roots yₖ of Chebyshev polynomial, x(yₖ) is uniform in (-1, 1).
+    x = jnp.arange(-deg + 1, deg + 1, 2) / deg
+    w = 2 / deg * jnp.ones(deg)
+    return x, w
+
+
+def chebgauss2(deg):
+    """Gauss-Chebyshev quadrature of the second kind.
+
+    Returns quadrature points xₖ and weights wₖ for the approximate evaluation
+    of the integral ∫₋₁¹ f(x) dx ≈ ∑ₖ wₖ f(xₖ) where f(x) = g(x) √(1−x²).
+
+    Parameters
+    ----------
+    deg : int
+        Number of quadrature points.
+
+    Returns
+    -------
+    x, w : (jnp.ndarray, jnp.ndarray)
+        Shape (deg, ).
+        Quadrature points and weights.
+
+    """
+    # Adapted from
+    # github.com/scipy/scipy/blob/v1.14.1/scipy/special/_orthogonal.py#L1803-L1851.
+    t = jnp.arange(deg, 0, -1) * jnp.pi / (deg + 1)
+    x = jnp.cos(t)
+    w = jnp.pi * jnp.abs(jnp.sin(t)) / (deg + 1)
```

Comment:
shouldn't this be sin^2(t)

Reply:
abs(sin(t)) = sqrt(1-x^2). This is useful for quadrature where that weight factor is not separable from the function to integrate. in our case f = g sqrt(1-lambda B)

Reply:
i am saying that the extra sin factor is buried inside the function to integrate

Reply:
To be honest, I would prefer the proper definition of the quadrature weights instead of yours which depends on the function to integrate. Or at least state the specific use case of this utility function in docstring such that in the future we don't forget why it is how it is.
What I mean is someone may misuse this function without knowing your implementation case.

Reply:
The docstring mentions
> Returns quadrature points xₖ and weights wₖ for the approximate evaluation
    of the integral ∫₋₁¹ f(x) dx ≈ ∑ₖ wₖ f(xₖ) where f(x) = g(x) √(1−x²).

Note that this quadrature is different from  ∫₋₁¹ √(1−x²) f(x) dx ≈ ∑ₖ wₖ f(xₖ) where f(x) = g(x). The latter would fail for bounce integrals. I can change the name of the function to your preference, but I cannot change the definition of the weights

Reply:
Ah ok, I see now.

> Note that this quadrature is different from ∫₋₁¹ √(1−x²) f(x) dx ≈ ∑ₖ wₖ f(xₖ) where f(x) = g(x).

I confused this with the original definition. My bad.